Lib/test/test_os.py

```diff
@@ -835,6 +835,31 @@ def test_key_type(self):
         self.assertIs(cm.exception.args[0], missing)
         self.assertTrue(cm.exception.__suppress_context__)
 
+    def _test_environ_iteration(self, iterator):
+        new_key = "__new_key__"
+
+        next(iterator)  # start iteration over os.environ.items
+
+        # add a new key in os.environ mapping
+        os.environ[new_key] = "test_environ_iteration"
+
+        next(iterator)  # force iteration over modified mapping
+        self.assertEqual(os.environ[new_key], "test_environ_iteration")
+
+        del os.environ[new_key]
+
+    def test_iter_error_when_changing_os_environ(self):
+        iter_environ = iter(os.environ)
+        self._test_environ_iteration(iter_environ)
+
+    def test_iter_error_when_changing_os_environ_items(self):
+        iter_environ_items = iter(os.environ.items())
+        self._test_environ_iteration(iter_environ_items)
+
+    def test_iter_error_when_changing_os_environ_values(self):
+        iter_environ_values = iter(os.environ.values())
+        self._test_environ_iteration(iter_environ_values)
```

Comment:
Why repeat "iter environ values" so much times? Why not inline the `iter_environ_values` variable?